nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -245,7 +268,14 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     curvature = 2.0 * carrot_pose.pose.position.y / carrot_dist2;
   }
 
-  linear_vel = desired_linear_vel_;
+  // Setting the velocity direction
+  double sign = 1.0;
+
+  if (allow_reversing_) {
+    sign = carrot_pose.pose.position.x >= 0.0 ? 1.0 : -1.0;
+  }
+
+  linear_vel = sign * desired_linear_vel_;
```

Comment:
Why set linear velocity here at all? Won't everything just be fabs-ed in the applyConstraint function? Then you could remove the fabs-ing you did in that function and just apply the `sign` when required

Reply:
You could largely revert all of the changes you made in applyConstraints before L489